### DISCUSSION TITLE
Register ts-node dynamically in ESM project?

### DISCUSSION THREAD

**Discussion by B4nan** [TypeStrong/ts-node#1880] [Q&A] [answered]
Context: I am working on MikroORM, where I dynamically load JS/TS files (let it be the ORM config or the entities). 

Trying to gradually improve the ESM support, and one thing I failed to get around is dynamic import of TS files. For CJS projects, I am just using `require()`, and I do register `ts-node` dynamically via `require('ts-node').register({ ... })`. This works fine, but fails for ESM projects, where the dynamic imports happen via `import()` instead. 

I can make it work with the `NODE_OPTIONS=\"--loader ts-node/esm\"` env var - is there a way to achieve this programatically, from inside the ORM CLI script? I read that the ESM support in ts-node works by forking the node script and using the loader, so I guess it is the best what we can do?

I would love to do something like `require('ts-node').register({ esm: true })` (which does not help).

**Comment by cspotcode** [comment, marked as answer]
Yeah this is a limitation of nodejs.  You should ask https://github.com/nodejs/loaders
